src/modules/fancyzones/lib/GenericKeyHook.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <functional>
+#include "pch.h"
+
+template<int... keys>
+class GenericKeyHook
```

Comment:
I really like how you've refactored the hook classes! The only restriction we currently have now are those pesky `static` hHook/callback members, which prevent us from having multiple hooks of the same type. but that's not important to have right now, and we really don't want to introduce thunks or overcomplicate matters.

Reply:
Thanks! Yeah, if we decide that we need to do that, we'll figure it out.

Reply:
OTOH the simplest way would be to introduce another `size_t` template parameter which will serve as an ID into `static 
 std::unordered_map<size_t, std::function<callback_t>> callbacks;` or something.